tests/neg/missing-implicit1.check

```diff
@@ -16,12 +16,7 @@
    |
    |    import testObjectInstance.instances.traverseList
    |
--- Error: tests/neg/missing-implicit1.scala:23:42 ----------------------------------------------------------------------
+-- [E008] Not Found Error: tests/neg/missing-implicit1.scala:23:18 -----------------------------------------------------
 23 |    List(1, 2, 3).traverse(x => Option(x)) // error
-   |                                          ^
-   |no implicit argument of type testObjectInstance.Zip[Option] was found for an implicit parameter of method traverse in trait Traverse
-   |
-   |The following import might fix the problem:
-   |
-   |  import testObjectInstance.instances.zipOption
-   |
+   |    ^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
this new error is definitely less helpful - it seems like the compiler should improve search of extension methods with context parameters on the first clause before we can make this change, which is not ideal

Reply:
Agreed. Note that the problem will occur with `using` too (that is, irrespective of this PR), so indeed we might want to address it separately:
```scala
extension [A, B, G[_]](fa: F[A])(using Zip[G]) def traverse(f: A => G[B]): G[F[B]]
```

Reply:
its probably worth opening an issue for the manual desugaring not being found by the compiler

Reply:
same here, could the extension methods be rewritten so that the error does not change?